### PR TITLE
Mongo7 upgrade

### PR DESCRIPTION
**Pull request by Xiangs18**

### CODE REVIEW

**Comment by Xiangs18** [comment]
Hi, tests pass in CI, but the service container fails after upgrading to Mongo7. Still debugging to find the root cause.

**Comment by Xiangs18** [comment]
> Hi, tests pass in CI, but the service container fails after upgrading to Mongo7. Still debugging to find the root cause.

Gavin's [solution](https://kbase.slack.com/archives/C05SSJT13BN/p1721168581623819) fixed the bug.

File: src/jgikbase/test/idmapping/storage/mongo/test_id_mapping_mongo_storage.py [hunk 1]
```diff
@@ -168,9 +166,9 @@ def test_startup_with_2_config_docs(mongo):
 
     # pattern matcher for the error format across python 2 & 3
     p = re.compile(
-        "Failed to create index: E11000 duplicate key error (index|collection): "
-        + r"test_id_mapping.config( index: |\.\$)schema_1\s+dup key: "
-        + r'\{ : "schema" \}'
+        r'Failed to create index: .*?'
+        r'E11000 duplicate key error collection: .*?'
+        r' index: schema_1 dup key: .*?'
```
Comment (MrCreosote) [inline src/jgikbase/test/idmapping/storage/mongo/test_id_mapping_mongo_storage.py:171]: This no longer tests that the db and collection names are in the error message

Same for the duplicate key

File: src/jgikbase/idmapping/builder.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def _build_storage(self) -> IDMappingStorage:
                 client.admin.command("ismaster")
             except ConnectionFailure as e:
                 raise IDMappingBuildException("Connection to database failed") from e
-            db = client[self.cfg.mongo_db]
+            db = client[self.cfg.mongo_db]  # type: ignore
             self._storage: IDMappingStorage = IDMappingMongoStorage(db)
```
[the remark this answers is not in the document]
Comment (Xiangs18) [reply src/jgikbase/idmapping/builder.py:101]: retryWrites is in the next PR.

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/idmapping/builder.py:101]: :+1:

File: src/jgikbase/test/idmapping/storage/mongo/test_id_mapping_mongo_storage.py (hunk 1 above)
Comment (Xiangs18) [reply src/jgikbase/test/idmapping/storage/mongo/test_id_mapping_mongo_storage.py]: 👍

File: src/jgikbase/test/idmapping/test_utils.py [hunk 3]
```diff
@@ -76,6 +76,11 @@ def find_free_port() -> int:
         return s.getsockname()[1]
 
 
+def remove_ns_from_index_info(index_info: dict) -> None:
+    for _, index_detail in index_info.items():
+        index_detail.pop("ns", None)
```
[the remark this answers is not in the document]
Comment (Xiangs18) [reply src/jgikbase/test/idmapping/test_utils.py]: 👍

File: src/jgikbase/test/idmapping/mongo_controller.py [hunk 4]
```diff
@@ -52,6 +52,9 @@ def __init__(
         if not root_temp_dir:
             raise ValueError("root_temp_dir is None")
 
+        # semver parser
+        s = semver.VersionInfo.parse
+
```
[the remark this answers is not in the document]
Comment (Xiangs18) [reply src/jgikbase/test/idmapping/mongo_controller.py]: In commit [8bcccd5](https://github.com/jgi-kbase/IDMappingService/pull/140/commits/8bcccd545aca3dd4ad188c864cb24869ded971be), `from semver.VersionInfo import parse as sver` failed.
So I removed `s = semver.VersionInfo.parse`, and use directly ...

File: src/jgikbase/idmapping/service/mapper_service.py [hunk 5]
```diff
@@ -207,14 +207,15 @@ def format(self, record):
...
             "source": record.name,
-            "ip": flask_req_global.ip,
-            "method": flask_req_global.method,
-            "callid": flask_req_global.req_id,
             "msg": record.getMessage(),
         }
         # https://docs.python.org/3.6/library/sys.html#sys.exc_info
         if record.exc_info and record.exc_info != (None, None, None):
             log["excep"] = _format_exception(record.exc_info[1])
+        if flask_req_global:
```
[the remark this answers is not in the document]
Comment (Xiangs18) [reply src/jgikbase/idmapping/service/mapper_service.py:215]: If my understanding is correct, logging is not yet tested in integration test? https://github.com/jgi-kbase/IDMappingService/blob/3f2c29312c329bb7deddea822f55c9c232b9d50c/src/jgikbase/test/idmapping/integration/service_test.py#L35

When I tested in the container service, these values did show up properly.

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/idmapping/service/mapper_service.py:215]: There's tests in mapper_service_test.py, you should add a test for the case where there's no request context.

File: src/jgikbase/test/idmapping/mongo_controller.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/test/idmapping/mongo_controller.py]: Ok, then I would do
```
import semver
sver = semver.VersionInfo.parse
```

File: src/jgikbase/test/idmapping/test_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/test/idmapping/test_utils.py]: :+1:

File: src/jgikbase/test/idmapping/storage/mongo/test_id_mapping_mongo_storage.py (hunk 1 above)
Comment (MrCreosote) [reply src/jgikbase/test/idmapping/storage/mongo/test_id_mapping_mongo_storage.py]: :+1:

File: src/jgikbase/test/idmapping/mongo_controller.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Xiangs18) [reply src/jgikbase/test/idmapping/mongo_controller.py]: 👍

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/test/idmapping/mongo_controller.py]: :+1:

File: src/jgikbase/idmapping/service/mapper_service.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Xiangs18) [reply src/jgikbase/idmapping/service/mapper_service.py:215]: 👍

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/idmapping/service/mapper_service.py:215]: :+1:

**Review by MrCreosote — APPROVED** [review]

**Pull request merged**